### PR TITLE
Provide env var for archive directory

### PR DESCRIPTION
**Pull request by lparsons**
## Summary Change Description

Provides the ability to configure the archive directory using and environment variable

## Affected Issues/Pull Requests

- Resolves #__issue_number__

- Partially addresses #732

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [x] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: TraceBase/settings.py [hunk 1]
```diff
@@ -142,14 +142,12 @@
...
 # https://docs.djangoproject.com/en/3.2/howto/static-files/
-
 STATIC_URL = "/static/"
-
 STATICFILES_DIRS = [os.path.join(BASE_DIR, "static")]
 
 # File storage location
 MEDIA_URL = "/archive/"
-MEDIA_ROOT = os.path.join(BASE_DIR, "archive")
+MEDIA_ROOT = env.str("ARCHIVE_DIR", default=os.path.join(BASE_DIR, "archive"))
```
Comment (hepcat72) [inline TraceBase/settings.py:150]: Q. I'm a little confused.  The end result of the default here is different from before.  So, before this change, the default `MEDIA_ROOT` would be `{BASE_DIR}/archive`.  After this change, given the value in `.env.example`, the default would be just `archive` (i.e. a relative path compared to the prepended `BASE_DIR` that was formerly applied).

I understand that the intent is to allow an absolute path in the env variable, but it seems confusing that if we don't define `ARCHIVE_DIR`, we would get `{BASE_DIR}/archive`, but get something different if we set `ARCHIVE_DIR` to "archive".

I know it's effectively the same, but why were we prepending `BASE_DIR` before?  Should we be changing whether `ARCHIVE_DIR` has an absolute or relative path and conditionally prepend `BASE_DIR` based on the result?

**Review by hepcat72 — COMMENTED** [review]
I'm a little confused about the intended result if `ARCHIVE_DIR` is set to a relative path.  Before, it was prepended with `BASE_DIR`, but now it's not (but it is if `ARCHIVE_DIR` is not defined).

File: TraceBase/settings.py (hunk 1 above)
Comment (hepcat72) [reply TraceBase/settings.py:150]: ...or maybe just don't prepend `BASE_DIR` in the default?

**Review by hepcat72 — APPROVED** [review]
This looks good.  Might be nice to specify that an absolute path is required, but the example implies it.

I also assume that if the value is ever uncommented/changed, moving existing files to the new location would preserve them without issue?

**Comment by lparsons** [comment]
> This looks good. Might be nice to specify that an absolute path is required, but the example implies it.

Oh, I didn't realize that an absolute path was required. Did you see that documented somewhere, or did you just try it? I can include some guidance if we can confirm it.

> I also assume that if the value is ever uncommented/changed, moving existing files to the new location would preserve them without issue?

Yes and no. Permissions and other issues could affect how things work, but someone wanting to do that should consult the Django docs and/or tread carefully. We are simply providing the ability to set a variable in the Django `settings.py` file here, it's up to admins to make sure they know what they are doing when then set it, imo.

**Comment by hepcat72** [comment]
> > This looks good. Might be nice to specify that an absolute path is required, but the example implies it.
> 
> Oh, I didn't realize that an absolute path was required. Did you see that documented somewhere, or did you just try it? I can include some guidance if we can confirm it.

Well, I tentatively inferred that in my first review, and then felt that your confirmation that the default was problematic, confirmed my prior inference.  I don't know if an absolute path is required.  I had just noted the prepend of `BASE_DIR` and didn't expand the diff upward to see how `BASE_DIR` was set.  If not prepending `BASE_DIR` is fine, then no problem.

> > I also assume that if the value is ever uncommented/changed, moving existing files to the new location would preserve them without issue?
> 
> Yes and no. Permissions and other issues could affect how things work, but someone wanting to do that should consult the Django docs and/or tread carefully. We are simply providing the ability to set a variable in the Django `settings.py` file here, it's up to admins to make sure they know what they are doing when then set it, imo.

Got it.  Referencing that in the comment would be icing, but not necessary.

**Pull request merged**